virttest/qemu_devices/qcontainer.py

```diff
@@ -1667,7 +1667,7 @@ def cdroms_define_by_params(self, name, image_params, media=None,
         image_params['image_raw_device'] = 'yes'
         cd_format = image_params.get('cd_format')
         scsi_hba = image_params.get("scsi_hba")
-        if cd_format is None or cd_format is 'ide':
+        if cd_format in (None, "scsi-cd", "scsi-cd"):
```

Comment:
Typo here, should be `if cd_format in (None, "ide", "scsi-cd")`, right?

Reply:
I don't think `"scsi-cd"` should in the tuple. 
if it is, why do you need this:
`if machine_type == "pseries":` `cd_format = "scsi-cd"`

Reply:
@luckyh: Yes, it should be `None, "ide", "scsi-cd"`, I'll fix it in v2.
@hereischen: as mentioned in commit message, it already works well for cds defined by cart config. But when the device is added by test (either cfg or in code) the `if machine_type == "pseries"` might not be executed as the cd_format is already something else. Although it's important to override the other important variables like `scsi_hbas` which are only corrected when cd_format is `ide` or `None` but when the device is added in test the `cd_format` might be different, while the `hbas` etc might not be set by cartesian.